### PR TITLE
Feature/item dependency display it to regular users

### LINKED ISSUE #1068: Item dependency: display it to regular users
## Overview

Display to users what they need to solve to unlock access to content.

## Acceptance criteria
- [ ] When a user has "can_view=info" on an item (for instance, user 456 on this [task](https://dev.algorea.org/en/#/activities/by-id/4297212147988794347;path=4702,1625159049301502151;parentAttempId=0/details) or [this chapter](https://dev.algorea.org/en/#/activities/by-id/128139237432513103;path=4702,4102;parentAttempId=0/details)), [fetch the prerequisities](https://franceioi-algorea.s3.eu-west-3.amazonaws.com/spec/v1.2.0.html#operation/itemPrerequisitesView) of this item
- [ ] If this list is empty, keep the same behavior as now
- [ ] If the returned list in not empty, display **after** the current message ("Your current access rights do not allow you to start the activity"):
  - [ ] with the same style as the previous message "You can unlock access by validating one of the following content."
  - [ ] the list of prerequisites, with the score in front like in the children list and the lock on the right (but no "order" in front on them, no "category" state widget)
  - [ ] link to prerequisite are pathless, display the path suggestion on hover
- [ ] When a user has "can_view=none" or "can_view > info", there is no changes

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #1068

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

...

## Test cases

- [ ] Case 1:
  1. Given I am the user 456
  2. When I go to [this page](https://dev.algorea.org/branch/feature/item-dependency-display-it-to-regular-users/en/activities/by-id/4297212147988794347;path=4702,1625159049301502151;parentAttempId=0/details)
  3. Then I see the message "Your current access rights do not allow you to start the activity." in new design and list of dependency
  4. And I hover on "Plateforme de démonstration"
  5. Then I see tooltip with path suggestion

- [ ] Case 2:
  1. Given I am the user 456
  2. When I go to [this page](https://dev.algorea.org/branch/feature/item-dependency-display-it-to-regular-users/en/activities/by-id/128139237432513103;path=4702,4102;parentAttempId=0/details)
  3. Then I see the message "Your current access rights do not allow you to list the content of this chapter." in new design and list of dependency

### CODE REVIEW

File: src/app/modules/item/components/item-unlock-access/item-unlock-access.component.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  private readonly itemId$ = new ReplaySubject<string>(1);
+  private readonly refresh$ = new Subject<void>();
+
+  state$ = this.itemId$.pipe(
+    switchMap(itemId => this.getItemPrerequisitesService.get(itemId)),
+    mapToFetchState({ resetter: this.refresh$ }),
+    share(),
+  );
+  addedIds$ = this.state$.pipe(readyData(), map(data => data.map(dependency => dependency.id)));
```
Comment (smadbe) [inline src/app/modules/item/components/item-unlock-access/item-unlock-access.component.ts:29]: what is this, it seems unused

Comment (Iloveall) [reply src/app/modules/item/components/item-unlock-access/item-unlock-access.component.ts]: Fixed

File: src/app/shared/models/domain/item-view-permission.ts [hunk 2]
```diff
@@ -25,6 +25,10 @@ export function allowsViewingInfo(p: ItemPermWithView): boolean {
   return [ P.Info, P.Content, P.ContentWithDescendants, P.Solution ].includes(p.canView);
 }
 
+export function allowsViewingOnlyInfo(p: ItemPermWithView): boolean {
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/shared/models/domain/item-view-permission.ts]: Fixed

**Review by smadbe — APPROVED** [review]
Looks good, thx 👍

**Pull request merged**